### PR TITLE
deprecate obsreport.Must functions in favor of obsreport.New

### PR DESCRIPTION
**Pull request by paivagustavo**
**Description:** Deprecate `obsreport.Must*` functions, make deprecated `obsreport.New*` functions to return an error and undeprecate them.

The `ScraperControllerReceiver` was changed to created a `obsreport.Scraper` for each configured scraper during the `NewScraperControllerReceiver` instead of creating a new `obsreport.Scraper` every scrape.

**Link to tracking Issue:** https://github.com/open-telemetry/opentelemetry-collector/issues/6458

### CODE REVIEW

File: .chloggen/return_error_for_obsreport_new_functions.yaml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: 'deprecation'
+
+# The name of the component, or a single word describing the area of concern, (e.g. otlpreceiver)
+component: obsreport
+
+# A brief description of the change.  Surround your text with quotes ("") if it needs to start with a backtick (`).
+note: "`obsreport.New[Receiver|Scraper|Processor|Exporter]` returns error now, deprecate `obsreport.MustNew` methods"
```
Comment (bogdandrutu) [inline .chloggen/return_error_for_obsreport_new_functions.yaml:8]: We need 2 "chloggen" entries (that's what we do in this case):
1. deprecation -> Must*
2. breaking -> change signature of already deprecated

File: obsreport/obsreport_exporter.go [hunk 2]
```diff
@@ -64,19 +64,14 @@ type ExporterSettings struct {
...
-func NewExporter(cfg ExporterSettings) *Exporter {
-	exp, err := newExporter(cfg, featuregate.GetRegistry())
-	if err != nil && cfg.ExporterCreateSettings.Logger != nil {
-		cfg.ExporterCreateSettings.Logger.Warn("Error creating an obsreport.Exporter", zap.Error(err))
-	}
-
-	return exp
+// NewExporter creates a new Exporter.
+func NewExporter(cfg ExporterSettings) (*Exporter, error) {
+	return newExporter(cfg, featuregate.GetRegistry())
```
Comment (bogdandrutu) [inline obsreport/obsreport_exporter.go:69]: Don't need the private one. Just rename newExporter -> NewExporter

File: obsreport/obsreport_receiver.go [hunk 3]
```diff
@@ -76,18 +76,14 @@ type ReceiverSettings struct {
...
-// Deprecated: [v0.64.0] use MustNewReceiver.
-func NewReceiver(cfg ReceiverSettings) *Receiver {
-	rcv, err := newReceiver(cfg, featuregate.GetRegistry())
-	if err != nil && cfg.ReceiverCreateSettings.Logger != nil {
-		cfg.ReceiverCreateSettings.Logger.Warn("Error creating an obsreport.Receiver", zap.Error(err))
-	}
-	return rcv
+// NewReceiver creates a new Receiver.
+func NewReceiver(cfg ReceiverSettings) (*Receiver, error) {
+	return newReceiver(cfg, featuregate.GetRegistry())
```
Comment (bogdandrutu) [inline obsreport/obsreport_receiver.go:81]: Same as exporter

**Review by bogdandrutu — APPROVED** [review]

File: obsreport/obsreport_exporter.go (hunk 2 above)
Comment (paivagustavo) [reply obsreport/obsreport_exporter.go:69]: The private one is mostly to wrap the `featuregate` since it's only used for tests.

**Comment by bogdandrutu** [comment]
@paivagustavo please prepare a PR in contrib to update for the new code. Use `make update-otel` in contrib repo to upgrade to latest  merged PR.

File: .chloggen/return_error_for_obsreport_new_functions.yaml [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+# A brief description of the change.  Surround your text with quotes ("") if it needs to start with a backtick (`).
+note: "`obsreport.New[Receiver|Scraper|Processor|Exporter]` returns error now"
+
+# One or more tracking issues or pull requests related to the change
+issues: [6458]
+
+# (Optional) One or more lines of additional information to render under the primary note.
+# These lines will be padded with 2 spaces and then inserted directly into the document.
+# Use pipe (|) for multiline entries.
+subtext:
```
[the remark this answers is not in the document]
Comment (bogdandrutu) [reply .chloggen/return_error_for_obsreport_new_functions.yaml:16]: ping @paivagustavo

**Review by codeboten — APPROVED** [review]

**Pull request merged**